tests/signals/tests.py

```diff
@@ -626,3 +626,29 @@ async def failing_async_handler(**kwargs):
                 (async_handler, 1),
             ],
         )
+
+
+class OnlyAsyncReceiversTests(SimpleTestCase):
+    async def test_asend_with_only_async_receivers(self):
+        async_handler = AsyncHandler()
+        signal = dispatch.Signal()
+        signal.connect(async_handler)
+
+        try:
+            result = await signal.asend(self.__class__)
+        except TypeError:
+            self.fail("asend raised TypeError unexpectedly.")
+        else:
+            self.assertEqual(result, [(async_handler, 1)])
+
+    async def test_asend_robust_with_only_async_receivers(self):
+        async_handler = AsyncHandler()
+        signal = dispatch.Signal()
+        signal.connect(async_handler)
+
+        try:
+            result = await signal.asend_robust(self.__class__)
+        except TypeError:
+            self.fail("asend_robust raised TypeError unexpectedly.")
+        else:
+            self.assertEqual(result, [(async_handler, 1)])
```

Comment:
We normally don't translate errors to fails in tests
```suggestion
        result = await signal.asend_robust(self.__class__)
        self.assertEqual(result, [(async_handler, 1)])
```

Reply:
Thank you, I've made the changes in [`9433ce1` (#17837)](https://github.com/django/django/pull/17837/commits/9433ce1fd8f24f88a3fd2a37f918e8bf46115c01).